src/CodeGen_Hexagon.cpp

```diff
@@ -177,6 +177,7 @@ void CodeGen_Hexagon::init_module() {
     HvxIntrinsic intrinsic_wrappers[] = {
         // Zero/sign extension:
         { IPICK(Intrinsic::hexagon_V6_vzb), u16x2,  "zxt.vub", {u8x1} },
+        { IPICK(Intrinsic::hexagon_V6_vzb), i16x2,  "zxt.vub", {u8x1} },
```

Comment:
I don't think this is actually needed, you just need the new patterns in HexagonOptimize.cpp. These two intrinsic declarations should generate the exact same LLVM functions.